pandas/tests/test_strings.py

```diff
@@ -8,10 +9,13 @@
 from pandas._libs import lib
 
 from pandas import DataFrame, Index, MultiIndex, Series, concat, isna, notna
+from pandas.core.base import PandasObject
 import pandas.core.strings as strings
 import pandas.util.testing as tm
 from pandas.util.testing import assert_index_equal, assert_series_equal
 
+index_or_series_params = [Index, Series]  # type: List[Type[PandasObject]]
```

Comment:
With this, I get

```
pandas/tests/test_strings.py:18: error: Type variable "pandas._typing.IndexOrSeries" is unbound
```

Any suggestions?

Reply:
probably don't need to create IndexOrSeries in _typing. so the union or the shared class is fine, but should probably be consistent.

Reply:
@simonjayhawkins do you know generally how mypy is inferencing these? I guess from the original list of errors I'm not sure where `Type[PandasObject]` even comes into play. Wondering if there isn't a core issue we are missing here, but I may also just be overlooking

Reply:
my first reaction is that it is a mypy error. `Revealed type is 'builtins.list[def (data: Any =, Any =, Any =, name: Any =, Any =, Any =, **Any) -> pandas.core.base.PandasObject]'` seems iffy.

but agree, that it may be better to try and determine the cause rather than making these changes.

Reply:
That `reveal_type` trick is handy. It'd be nice if there was a way to print debug info about a particular chain of inference: all the decisions that led to mypy inferring that type.

Reply:
I wonder if types are getting mangled from pytest

Reply:
i don't think pytest is relevant...

```
from typing import List, Type, cast

import pandas as pd

reveal_type([pd.Index, pd.Series])

[pd.Index, pd.Series]

cast(List[Type[pd.core.base.PandasObject]], [pd.Index, pd.Series])
```

```
$ mypy test.py
test.py:5: note: Revealed type is 'builtins.list[def (data: Any =, Any =, Any =, name: Any =, Any =, Any =, Any =) -> pandas.core.base.PandasObject]'
test.py:5: error: List item 0 has incompatible type "Type[Index]"; expected "Type[PandasObject]"
test.py:7: error: List item 0 has incompatible type "Type[Index]"; expected "Type[PandasObject]"
(pandas-dev) 
```

Reply:
i don't think the `reveal_type([pd.Index, pd.Series])` should be giving an error which is why i suspect a mypy issue.

Reply:
Hmmm.....so when I run the above this is all I get:

```sh
$ mypy -V
mypy 0.720
$ mypy test.py
test.py:5: note: Revealed type is 'builtins.list[builtins.type*]'
```

Wonder why we get different output.

Reply:
need to clone #28394